index-handlers/src/test/java/no/sikt/nva/nvi/index/apigateway/SearchNviCandidatesHandlerIntegrationTest.java

```diff
@@ -190,6 +210,115 @@ void shouldFilterByReportingPeriodAndNotPublicationYear() {
+    private static final String UNRELATED_ASSIGNED = "assignedAndUnrelated";
+    private static final String UNRELATED_UNASSIGNED = "unassignedAndUnrelated";
+
+    private static final Map<String, NviCandidateIndexDocument> docs =
+        createDocumentsForAssigneeTests();
+
+    @BeforeEach
+    void beforeEach() {
+      addDocumentsToIndex(docs.values());
+    }
+
+    @ParameterizedTest
+    @MethodSource("assigneeTestCaseProvider")
+    void shouldFilterByAssigneeAndExcludeUnassignedFilter(
+        Map<String, String> queryParameters, Collection<String> expectedDocumentKeys) {
+      var response = handleRequest(queryParameters);
+
+      assertThat(response.getHits())
+          .hasSameSizeAs(expectedDocumentKeys)
+          .allSatisfy(
+              doc -> {
+                var actualTitle = doc.publicationDetails().title();
+                assertThat(doc.id()).isEqualTo(docs.get(actualTitle).id());
+                assertThat(actualTitle).isIn(expectedDocumentKeys);
+              });
+    }
+
+    private static NviCandidateIndexDocument documentWithApprovals(
+        String title, Approval... approvals) {
+      var allApprovals = List.of(approvals);
+      var topLevelOrganizations = allApprovals.stream().map(Approval::institutionId).toList();
+      var details = randomPublicationDetailsBuilder(topLevelOrganizations).withTitle(title).build();
+      return randomIndexDocumentBuilder(details)
+          .withApprovals(allApprovals)
+          .withNumberOfApprovals(allApprovals.size())
+          .build();
+    }
+
+    private static Map<String, NviCandidateIndexDocument> createDocumentsForAssigneeTests() {
+      var assignedToCurrentUser = randomApproval(OUR_USER, OUR_ORGANIZATION);
+      var assignedToOtherUser = randomApproval(OUR_OTHER_USER, OUR_ORGANIZATION);
+      var assignedToTheirUser = randomApproval(THEIR_USER, THEIR_ORGANIZATION);
+      var noAssigneeFromUs = randomApproval(null, OUR_ORGANIZATION);
+      var noAssigneeFromThem = randomApproval(null, THEIR_ORGANIZATION);
+
+      return Map.of(
+          ASSIGNED_TO_CURRENT,
+          documentWithApprovals(ASSIGNED_TO_CURRENT, assignedToCurrentUser, noAssigneeFromThem),
+          ASSIGNED_TO_OTHER,
+          documentWithApprovals(ASSIGNED_TO_OTHER, assignedToOtherUser, assignedToTheirUser),
+          UNASSIGNED_BY_US,
+          documentWithApprovals(UNASSIGNED_BY_US, noAssigneeFromUs, assignedToTheirUser),
+          UNASSIGNED_BY_THEM,
+          documentWithApprovals(UNASSIGNED_BY_THEM, assignedToOtherUser, noAssigneeFromThem),
+          UNASSIGNED_BY_ALL,
+          documentWithApprovals(UNASSIGNED_BY_ALL, noAssigneeFromUs, noAssigneeFromThem),
+          UNRELATED_ASSIGNED,
+          documentWithApprovals(UNRELATED_ASSIGNED, assignedToTheirUser),
+          UNRELATED_UNASSIGNED,
+          documentWithApprovals(UNRELATED_UNASSIGNED, noAssigneeFromThem));
+    }
+
+    private static Stream<Arguments> assigneeTestCaseProvider() {
```

Comment:
This is admittedly not the most readable test setup, but we need to test a wide set of combinations of assignee status for the user organization and other organizations. Suggestions are welcome.